gguf-py/gguf/gguf.py

```diff
@@ -616,18 +621,39 @@ class GGUFWriter:
     temp_file: tempfile.SpooledTemporaryFile[bytes] | None = None
     tensors: list[tuple[np.ndarray[Any, Any], int]]
 
-    def __init__(self, path: os.PathLike[str] | str, arch: str, use_temp_file = True):
+    def get_pack_prefix(self):
```

Comment:
> I think this requires to bump the GGUF version because the current spec is explicit in little endianess. The spec should also be updated to reflect this change. We cannot simply trust that people do not distribute big endian files.
> 
> And of course bump the package version in `pyproject.toml`

> I think this requires to bump the GGUF version because the current spec is explicit in little endianess. The spec should also be updated to reflect this change. We cannot simply trust that people do not distribute big endian files.
> 
> And of course bump the package version in `pyproject.toml`

@monatis @ggerganov if people is goting to use big endian GGUF file on little endian system, they will get incorrect magic code error. Maybe we can use magic code to warn user that they are using wrong endianess file

```shell
> ./main -m ~/gguf-s390/Baichuan-7B-f16.gguf
Log start
main: build = 1359 (e4efbdb)
main: built with cc (SUSE Linux) 7.5.0 for x86_64-suse-linux
main: seed  = 1697012999
gguf_init_from_file: invalid magic number 47475546
error loading model: llama_model_loader: failed to load model from /home/cqy/gguf-s390/Baichuan-7B-f16.gguf

llama_load_model_from_file: failed to load model
llama_init_from_gpt_params: error: failed to load model '/home/cqy/gguf-s390/Baichuan-7B-f16.gguf'
main: error: unable to load model
```